addons/purchase/models/purchase.py

```diff
@@ -972,7 +972,6 @@ def onchange_product_id(self):
             return
 
         # Reset date, price and quantity since _onchange_quantity will provide default values
-        self.date_planned = self.order_id.date_planned or self._convert_to_middle_of_day(datetime.today())
```

Comment:
I have tested the flow, the fix is ok. 
But if you have 2 products (A and B) with each the same vendor but different lead time (5 for product A and 10 days for product B):
- Create a PO with the product vendor
- Add product A in first PO line, then PO line date_planned = PO date_planned = today + 5 day
- Add product B in a second line, then the  PO line date_planned will be the PO date_planned (= to the first PO line)
For the second line is seems weird no (the order of line create matter)? it shouldn't be the today + 10 day (using it leads day instead of the PO date planned) ?